src/openshift/url.ts

```diff
@@ -22,11 +23,60 @@ export class Url {
         } else if (ports.length > 1) {
             port = await vscode.window.showQuickPick(ports, {placeHolder: "Select port to expose"});
         } else {
-            return Promise.reject(`Component '${context.getName()}' has no ports decalred.`);
+            return Promise.reject(`Component '${data.getName()}' has no ports decalred.`);
         }
         return port === undefined ? undefined : Promise.resolve()
-            .then(async () => Url.odo.execute(Command.createCompontentUrl(project.getName(), app.getName(), context.getName(), port)))
-            .then(() => `URL for component '${context.getName()}' successfully created`)
-            .catch((err) => Promise.reject(`Failed to create URL for component '${context.getName()}'`));
+            .then(async () => Url.odo.execute(Command.createCompontentUrl(project.getName(), app.getName(), data.getName(), port)))
+            .then(() => `URL for component '${data.getName()}' successfully created`)
+            .catch((err) => Promise.reject(`Failed to create URL for component '${data.getName()}'`));
+    }
+
+    static async getUrlData(url) {
+        let urlName: OpenShiftObject  = url ? url : undefined;
+        let name: OpenShiftObject;
+        if (!urlName) {
+            name = await Url.getProjectName();
+        }
+        if (urlName) {
+            urlName = url;
+        }
+        return name || urlName ? name || urlName : undefined;
+    }
+
+    static async getProjectName() {
```

Comment:
I have a fundamental issue with the way these helper methods are being added. From what I've seen the same ~4 methods appear in about 5 pull requests, each time in a different class.

Copy-paste on this level is one of the worst coding practices (really, it is). In order to materialize some more elegant solution, I've created #499 with a proposal on how to proceed. Feel free to chime in and discuss potential solutions. 
But this code is not an acceptable way forward.